pkg/skaffold/event/v2/event.go

```diff
@@ -173,6 +180,9 @@ func (ev *eventHandler) forEach(listeners *[]*listener, log *[]proto.Event, lock
 }
 
 func (ev *eventHandler) forEachEvent(callback func(*proto.Event) error) error {
+	if len(handler.wait) != cap(handler.wait) {
```

Comment:
you can run into a race condition from multiple goroutines and have a deadlock here.

Reply:
Updated this to use a select statement instead